src/main/java/com/sumologic/log4j/SumoLogicAppender.java

```diff
@@ -234,6 +239,12 @@ public void append(LogEvent event)
 
     try {
       queue.add(message);
+      if (event.getLevel().isInRange(Level.FATAL, Level.ERROR)) {
+        if(!queue.hasError()) {
+          queue.setError(true);
+          this.flusher.interrupt();
```

Comment:
This seems dangerous and overall looks really racy, why wouldn't the queue have picked up the event between the `add` call and here? 

It makes more sense to me to have the "timer" that is delaying between sends to be shorted to 0 as a first step here instead of trying to manage an error state flag. Such a setup might require re-configuring how the delay happens.

An alternative approach would be to have a priority queue for the events being pushed out, but that seems like a lot of overhead.